### ISSUE TITLE
Tab control accent colour management is broken/convoluted

### ISSUE THREAD

**Issue by bdach** [ppy/osu#8117]
Spotted during review of #7993.

Accent colour management for tab controls is all over the place. The root issue is that `OsuTabControl.AccentColour` setter's logic is flawed:

https://github.com/ppy/osu/blob/b41e10e001ad4283d950f6f7bca2695a90262c1f/osu.Game/Graphics/UserInterface/OsuTabControl.cs#L26-L38

If a tab control is added past calling this setter then it will not have its proper colour assigned. This leads to a bunch of places where this was retroactively fixed up by passing colours explicitly in object initialisers to the tabs in `CreateTabItem(T)`. This has a lot of users and a lot of inheritance logic, so it is very hard for me to keep in mind what has what colour.

Right now I'm testing out making `OsuTabControl` implement something like this:

```csharp
    /// <summary>
    /// A component that provides an accent colour to a hierarchy of drawables.
    /// Aimed for dependency injection usage in complex drawable hierarchies where using <see cref="IHasAccentColour"/>
    /// or passing down accent colours is impractical or not readable.
    /// </summary>
    public interface IAccentColourProvider
    {
        public IBindable<Color4> AccentColour { get; }
    }
```

and having this injected near each component that needs the colour, but it might be a few days before I can even tell if this is a proper direction to take because of the sheer scope of changes required.

**Comment by smoogipoo** [comment]
Yeah, the `IHasAccentColour` interface sucks because it doesn't provide a bindable colour. [Other areas](https://github.com/ppy/osu/blob/master/osu.Game/Rulesets/Objects/Drawables/DrawableHitObject.cs#L32) in the game re-implement accent colour as bindables without implementing the interface to bypass this issue.

I'm not sure that DI-able accent colours are really necessary though. It sounds to me like the implementation complexity could be more trouble than its worth. Instead doing `AccentColour = { BindTarget = AccentColour }` whenever required sounds like the quickest solution. Doing it via DI starts making it hard to test these components, where you typically need some sort of "wrapping" test scene that provides the DI'd member.

It would be good to split out the accent colour as you mentioned.
